### PR TITLE
NEWSWORLDSERVICE-1732: Adds story snapshot delay and font block loading strategy

### PR DESCRIPTION
**Pull request by karinathomasbbc**
Resolves JIRA https://jira.dev.bbc.co.uk/browse/NEWSWORLDSERVICE-1732

Overall changes
======
To help fix flakiness with our Chromatic visual regression tests, we've added a delay to help with image loading and animations. We've also changed our font loading strategy from `optional` to `block`.

The `font-display: optional` [allows browsers to make a decision not to show the custom font based on connection speed](https://css-tricks.com/almanac/properties/f/font-display/#aa-values), to remove the potential flakiness of this, `block` has been used to give a short blocking time and infinite swap time which seems to fix font flakiness 🙏.

Adding the delay values does add additional time for Chromatic to generate and process snapshots, 5 seconds seems to be a good middle ground for loading everything we need to prevent flaky regression, and only increases Chromatic's snapshot and testing time by around about 40 seconds, compared to the max 15 seconds which increases the time by over 3 minutes. All finish before our E2E's finish anyway so this wouldn't affect our time to merge in PR's currently

15 Second Delay
image

5 Second Delay (And 8 seconds for live label pulse stories)
image

No delay
image

Code changes
======
- Adds a 5 second delay by default to all stories to cover image loading
- Adds an 8 second delay to stories with the Live label pulse to make sure animation has finished after DOM load
- Switches font loading strategy from `optional` to `block`
- Disables snapshotting of psammead-media-player story `With Callbacks` as it's an external iframe player with a constant live pulse flashing

Testing
======
1. Go through Chromatic visual regression UI and check to see fonts and live pulse are rendering consistently

Helpful Links
======
https://www.chromatic.com/docs/delay
https://css-tricks.com/almanac/properties/f/font-display/#aa-values

[Coding Standards](https://github.com/bbc/simorgh/blob/latest/docs/Coding-Standards/README.md)

[Repository use guidelines](https://github.com/bbc/simorgh-infrastructure/blob/latest/documentation/repository-guidelines.md)

### CODE REVIEW

**Comment by karinathomasbbc** [comment]
Sorry for snooping - this popped up as a GitHub notification in Slack. It got me thinking, while we're in here, could we also disable the visual snapshot for the Riddle Embed: https://latest--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/story/components-embeds-embed-html--riddle - you'll see it can't render, and if you've ever spotted this on your PR, the URL changes every time. Might as well exclude it when we can

**Comment by HarveyPeachey** [comment]
> Sorry for snooping - this popped up as a GitHub notification in Slack. It got me thinking, while we're in here, could we also disable the visual snapshot for the Riddle Embed: https://latest--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/story/components-embeds-embed-html--riddle - you'll see it can't render, and if you've ever spotted this on your PR, the URL changes every time. Might as well exclude it when we can



No it's fine, yeah definitely, I'll add it now

**Comment by HarveyPeachey** [comment]
> Sorry for snooping - this popped up as a GitHub notification in Slack. It got me thinking, while we're in here, could we also disable the visual snapshot for the Riddle Embed: https://latest--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/story/components-embeds-embed-html--riddle - you'll see it can't render, and if you've ever spotted this on your PR, the URL changes every time. Might as well exclude it when we can

I'm actually struggling with this a little bit, I'm not sure how to apply parameters to individual stories, I can't find any documentation on the API we use for [variant stories ](https://latest--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/story/coding-standards-storybook--page&knob-Promo%20Type=regular&knob-Select%20a%20service=news&knob-type=[object%20Object]#creating-story-for-a-component-with-variants)

**Comment by karinathomasbbc** [comment]
> > Sorry for snooping - this popped up as a GitHub notification in Slack. It got me thinking, while we're in here, could we also disable the visual snapshot for the Riddle Embed: https://latest--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/story/components-embeds-embed-html--riddle - you'll see it can't render, and if you've ever spotted this on your PR, the URL changes every time. Might as well exclude it when we can
> 
> I'm actually struggling with this a little bit, I'm not sure how to apply parameters to individual stories, I can't find any documentation on the API we use for [variant stories ](https://latest--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/story/coding-standards-storybook--page&knob-Promo%20Type=regular&knob-Select%20a%20service=news&knob-type=%5Bobject%20Object%5D#creating-story-for-a-component-with-variants)

Here's an example: https://github.com/bbc/simorgh/blob/857f0ba17f31d6b34b7821ed3307f256319d1de9/src/app/components/Copyright/index.stories.tsx#L31 - hope this helps

**Comment by HarveyPeachey** [comment]
> - hope this helps

That disables it for all the stories, we just want it for the Riddle sub story which I'm not sure how to do, currently seeing if I can find any docs around it

**Comment by karinathomasbbc** [comment]
> > * hope this helps
> 
> That disables it for all the stories, we just want it for the Riddle sub story which I'm not sure how to do, currently seeing if I can find any docs around it

I think you can use the same syntax to override just one story - try it out to see what happens

**Comment by HarveyPeachey** [comment]
> I think you can use the same syntax to override just one story - try it out to see what happens

I tried both of these and it won't render:
```jsx
export const Riddle = () => ({
  component: <EmbedHtml embeddableContent={riddleHTML} />,
  parameters: { chromatic: { disableSnapshot: true } },
});
```
```jsx
export const Riddle = {
  component: <EmbedHtml embeddableContent={riddleHTML} />,
  parameters: { chromatic: { disableSnapshot: true } },
};
```

Found some docs here mentioning [render](https://storybook.js.org/docs/writing-stories/stories-for-multiple-components#reusing-subcomponent-stories), so I tried that and it seems to work. Just need to see if Chromatic respects the parameters when I push it.

```jsx
export const Riddle = {
  render: () => <EmbedHtml embeddableContent={riddleHTML} />,
  parameters: { chromatic: { disableSnapshot: true } },
};
```

File: src/app/components/Embeds/EmbedHtml/index.stories.tsx [hunk 1]
```diff
@@ -21,6 +21,9 @@ export const IDT2QuotePic = () => (
   <EmbedHtml embeddableContent={quotePicEmbedHtml} />
 );
 
-export const Riddle = () => <EmbedHtml embeddableContent={riddleHTML} />;
+export const Riddle = {
+  render: () => <EmbedHtml embeddableContent={riddleHTML} />,
```
Comment (karinathomasbbc) [inline src/app/components/Embeds/EmbedHtml/index.stories.tsx:24]: Oh nice! Glad we managed to fix this 👍

**Review by karinathomasbbc — COMMENTED** [review]
Eek I thought I could approve this PR but I can't since I was the original author. But I approve!!

**Review by amoore108 — APPROVED** [review]

File: src/app/components/Embeds/EmbedHtml/index.stories.tsx (hunk 1 above)
Comment (Isabella-Mitchell) [reply src/app/components/Embeds/EmbedHtml/index.stories.tsx:25]: I think we still have these disabled on Chromatic (line 13). Is the intention to renable them?

File: src/app/components/Embeds/EmbedHtml/index.stories.tsx [hunk 2]
```diff
@@ -21,6 +21,9 @@ export const IDT2QuotePic = () => (
...
+  parameters: { chromatic: { disableSnapshot: true } },
```
Comment (Isabella-Mitchell) [inline src/app/components/Embeds/EmbedHtml/index.stories.tsx:25]: We have   `parameters: { chromatic: { disable: true } }` above on line 13 which I think applies to all these story files. Do we want to remove this from line 13? I'm not sure the difference between this and what you've added here for the Riddle component.

**Review by Isabella-Mitchell — APPROVED** [review]
Approving on behalf of myself and Karina - let's give it a spin!

File: src/app/components/Embeds/EmbedHtml/index.stories.tsx (hunk 2 above)
Comment (HarveyPeachey) [reply src/app/components/Embeds/EmbedHtml/index.stories.tsx:25]: > We have `parameters: { chromatic: { disable: true } }` above on line 13 which I think applies to all these story files. Do we want to remove this from line 13? I'm not sure the difference between this and what you've added here for the Riddle component.

I think this setting doesn't actually work, I've removed it

Comment (HarveyPeachey) [reply src/app/components/Embeds/EmbedHtml/index.stories.tsx:25]: Hmm it did work, I wonder why the riddle one was showing up then

**Pull request merged**